### PR TITLE
Main Sample Gen Script

### PR DESCRIPTION
**Pull request by theishshah**
This script will download and install the 1.0 operator-sdk binary and proceed to run the existing generate scripts for each of helm, go, and ansible samples.

### CODE REVIEW

File: .generate/gen-all.sh [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+HELM_SCRIPT_PATH="../helm/.generate/gen-helm-memcached.sh"
+GO_SCRIPT_PATH="../go/.generate/gen-go-sample.sh"
+ANS_SCRIPT_PATH="../ansible/.generate/gen-ansible-memcached.sh"
+
+RELEASE_VERSION=v1.0.0
+
+header_text "downloading binaries"
+
+curl -LO https://github.com/operator-framework/operator-sdk/releases/download/${RELEASE_VERSION}/operator-sdk-${RELEASE_VERSION}-x86_64-linux-gnu
```
Comment (estroz) [inline .generate/gen-all.sh:45]: Refactor these into a function:
```sh
function install_bin() {
	local url="https://github.com/operator-framework/operator-sdk/releases/download/${2}/${1}-${2}-x86_64-linux-gnu"
	curl -sSLo $1 $url
	chmod +x $1
	...
}

install_bin operator-sdk $RELEASE_VERSION
install_bin helm-operator $RELEASE_VERSION
install_bin ansible-operator $RELEASE_VERSION

...
```

File: .generate/gen-all.sh [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+curl -LO https://github.com/operator-framework/operator-sdk/releases/download/${RELEASE_VERSION}/ansible-operator-${RELEASE_VERSION}-x86_64-linux-gnu
+curl -LO https://github.com/operator-framework/operator-sdk/releases/download/${RELEASE_VERSION}/helm-operator-${RELEASE_VERSION}-x86_64-linux-gnu
+
+header_text "installing operator-sdk"
+
+chmod +x operator-sdk-${RELEASE_VERSION}-x86_64-linux-gnu && sudo mkdir -p /usr/local/bin/ && sudo cp operator-sdk-${RELEASE_VERSION}-x86_64-linux-gnu /usr/local/bin/operator-sdk && rm operator-sdk-${RELEASE_VERSION}-x86_64-linux-gnu
+chmod +x ansible-operator-${RELEASE_VERSION}-x86_64-linux-gnu && sudo mkdir -p /usr/local/bin/ && sudo cp ansible-operator-${RELEASE_VERSION}-x86_64-linux-gnu /usr/local/bin/ansible-operator && rm ansible-operator-${RELEASE_VERSION}-x86_64-linux-gnu
```
Comment (estroz) [inline .generate/gen-all.sh:52]: This script shouldn't need `sudo` for anything. Make a temp directory, move the binary there, and prepend that directory to `PATH`.

File: .generate/gen-all.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+chmod +x helm-operator-${RELEASE_VERSION}-x86_64-linux-gnu && sudo mkdir -p /usr/local/bin/ && sudo cp helm-operator-${RELEASE_VERSION}-x86_64-linux-gnu /usr/local/bin/helm-operator && rm helm-operator-${RELEASE_VERSION}-x86_64-linux-gnu
+
+HELM_OUTPUT=`"$HELM_SCRIPT_PATH"`
+echo $HELM_OUTPUT
```
Comment (estroz) [inline .generate/gen-all.sh:56]: Why not run these directly?
```suggestion
../helm/.generate/gen-helm-memcached.sh
```

**Review by estroz — CHANGES_REQUESTED** [review]

File: .generate/gen-all.sh (hunk 1 above)
Comment (camilamacedo86) [reply .generate/gen-all.sh:45]: IMO cannot be based for Linux. It needs work in Mac OS as well.

Comment (camilamacedo86) [reply .generate/gen-all.sh:45]: @estroz and @theishshah, 

The way that it is now it cannot work in a Mac. See my above suggestion to fix it.

**Comment by theishshah** [comment]
PLEASE REFER TO #162

**Pull request closed without merging**